### PR TITLE
chore: Fix colors examples

### PR DESCRIPTION
**Pull request by wainola**
## Related Issue Or Context

Closes: #133 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: examples/generic-node-example/package.json [hunk 1]
```diff
@@ -6,15 +6,17 @@
   "license": "MIT",
   "dependencies": {
     "ethers": "5.7.2",
-    "@buildwithsygma/sygma-sdk-core": "*",
+    "@buildwithsygma/sygma-sdk-core": "^1.1.4",
```
Comment (BeroBurny) [inline examples/generic-node-example/package.json:9]: Use monorepo version not external

File: examples/generic-node-example/tsconfig.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "jsx": "react-jsx"
   },
   "include": [
-    "src"
-  ]
+    "src",
+    "./.eslintrc.cjs"
```
Comment (BeroBurny) [inline examples/generic-node-example/tsconfig.json:25]: I think this should not be here

File: examples/generic-node-example/.eslintrc.cjs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  extends: "../../.eslintrc.cjs",
+  parserOptions: {
+    project: "./tsconfig.json",
+    tsconfigRootDir: __dirname,
+  },
+  rules: {
+    "@typescript-eslint/no-unsafe-member-access": "off",
+    "@typescript-eslint/no-unsafe-call": "off",
+  },
```
Comment (BeroBurny) [inline examples/generic-node-example/.eslintrc.cjs:10]: Please remove this linter disables

**Review by BeroBurny — CHANGES_REQUESTED** [review]

File: examples/generic-node-example/package.json (hunk 1 above)
Comment (wainola) [reply examples/generic-node-example/package.json:9]: monorepo version is `"*"`?

File: examples/generic-node-example/tsconfig.json (hunk 2 above)
Comment (wainola) [reply examples/generic-node-example/tsconfig.json:25]: removed

File: examples/generic-node-example/.eslintrc.cjs (hunk 3 above)
Comment (wainola) [reply examples/generic-node-example/.eslintrc.cjs:10]: removed

File: examples/generic-node-example/package.json (hunk 1 above)
Comment (wainola) [reply examples/generic-node-example/package.json]: I will rebase this PR with the latest from main that includes latest version of yarn and all examples are using monorepo version of sdk

**Review by enemycnt — CHANGES_REQUESTED** [review]
Please change type to commonjs for now, and fix the conflicts.

File: examples/generic-colors/package.json [hunk 4]
```diff
@@ -23,11 +23,12 @@
     "react-hook-form": "^7.30.0"
   },
   "scripts": {
-    "start": "PORT=3006 BROWSER=none react-app-rewired start",
+    "start": "DISABLE_ESLINT_PLUGIN=true PORT=3006 BROWSER=none react-app-rewired start",
```
Comment (BeroBurny) [inline examples/generic-colors/package.json:26]: remove eslint disable  flag

File: examples/generic-colors/src/App.tsx [hunk 5]
```diff
@@ -141,19 +149,21 @@ function App() {
...
-      });
-    } else {
-      dispatch({
-        type: "resetColorSelected",
-      });
-    }
-  };
+  const handleColorSelected =
+    (colorId: string) =>
+    ({ target: { value, checked } }: any) => {
```
Comment (BeroBurny) [inline examples/generic-colors/src/App.tsx:154]: please do not use `any`

File: examples/generic-node-example/.eslintrc.cjs (hunk 3 above)
Comment (BeroBurny) [inline examples/generic-node-example/.eslintrc.cjs:10]: Please remove this linter disables

File: examples/generic-node-example/src/depositGeneric.ts [hunk 6]
```diff
@@ -1,102 +1,141 @@
...
+  // Here we prepare to listen on destination chain
+  const signerDestinationChain = providerNode2.getSigner(account);
+
+  const filters = colorContractNode2
+    .connect(signerDestinationChain)
+    .filters.setColorEvent();
+
+  let counter = 0;
+
+  const listener = (color: string, ...rest: any): void => {
```
Comment (BeroBurny) [inline examples/generic-node-example/src/depositGeneric.ts:43]: please do not use `any`

File: examples/generic-node-example/src/depositGeneric.ts [hunk 7]
```diff
@@ -1,102 +1,141 @@
...
-      const colorDecoded = decodeColor(color)
-      console.log(`${Chalk.keyword('orange')('Color decoded')}: ${Chalk.hex(colorDecoded).bold(colorDecoded)}`)
-      colors.push(colorDecoded)
+    const iterable = Array.from(Array(colorArrayLength.toNumber()).keys()).map(
+      (i) => i
+    );
+
+    const colors = [];
+
+    for await (const k of iterable) {
```
Comment (BeroBurny) [inline examples/generic-node-example/src/depositGeneric.ts:74]: Please use more meaningful names for variable

File: examples/generic-node-example/tsconfig.json (hunk 2 above)
Comment (BeroBurny) [reply examples/generic-node-example/tsconfig.json:25]: unresolved as is still present

**Review by BeroBurny — CHANGES_REQUESTED** [review]

**Comment by wainola** [comment]
Update: this PR is stale since there is an issue with the usage of the internal dependency of the monorepo

**Comment by wainola** [comment]
Update: node version of colors example is not going to work until we resolve this [issue](https://github.com/sygmaprotocol/sygma-sdk/issues/196). Frontend examples is properly working with current version of local setup

File: examples/generic-colors/src/types/Colors.ts [hunk 8]
```diff
@@ -0,0 +1,345 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-colors/src/types/Colors.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-colors/src/types/common.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-colors/src/types/common.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-node-example/src/types/Colors.ts [hunk 10]
```diff
@@ -0,0 +1,345 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-node-example/src/types/Colors.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-node-example/src/types/ColorsAbi.ts [hunk 11]
```diff
@@ -0,0 +1,345 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-node-example/src/types/ColorsAbi.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-node-example/src/types/ColorsAbi__factory.ts [hunk 12]
```diff
@@ -0,0 +1,226 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-node-example/src/types/ColorsAbi__factory.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-node-example/src/types/Colors__factory.ts [hunk 13]
```diff
@@ -0,0 +1,218 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-node-example/src/types/Colors__factory.ts:3]: please do not use tslint disable and eslint disable

File: examples/generic-node-example/src/types/common.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
Comment (BeroBurny) [inline examples/generic-node-example/src/types/common.ts:3]: please do not use tslint disable and eslint disable

**Review by BeroBurny — CHANGES_REQUESTED** [review]
Notice naming for hooks is without prefixed `use` which actually tells developers its a hook
Please do this reading https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use

**Comment by BeroBurny** [comment]
> Update: node version of colors example is not going to work until we resolve this [issue](https://github.com/sygmaprotocol/sygma-sdk/issues/196). Frontend examples is properly working with current version of local setup

if there is an issue with node dependencies, then is better to separate this PR into 2 parts, browser, and node so one part can be merged and another when is blocker resolved. + This will help reviewers to review PR

**Review by enemycnt — DISMISSED** [review]

File: examples/generic-node-example/src/types/common.ts (hunk 14 above)
Comment (wainola) [reply examples/generic-node-example/src/types/common.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-node-example/src/types/Colors__factory.ts (hunk 13 above)
Comment (wainola) [reply examples/generic-node-example/src/types/Colors__factory.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-node-example/src/types/ColorsAbi__factory.ts (hunk 12 above)
Comment (wainola) [reply examples/generic-node-example/src/types/ColorsAbi__factory.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-node-example/src/types/ColorsAbi.ts (hunk 11 above)
Comment (wainola) [reply examples/generic-node-example/src/types/ColorsAbi.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-node-example/src/types/Colors.ts (hunk 10 above)
Comment (wainola) [reply examples/generic-node-example/src/types/Colors.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-colors/src/types/Colors.ts (hunk 8 above)
Comment (wainola) [reply examples/generic-colors/src/types/Colors.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-colors/src/types/ColorsAbi.ts [hunk 15]
```diff
@@ -0,0 +1,345 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/src/types/ColorsAbi.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-colors/src/types/Colors__factory.ts [hunk 16]
```diff
@@ -0,0 +1,218 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/src/types/Colors__factory.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-colors/src/types/common.ts (hunk 9 above)
Comment (wainola) [reply examples/generic-colors/src/types/common.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-colors/src/types/ColorsAbi__factory.ts [hunk 17]
```diff
@@ -0,0 +1,226 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/src/types/ColorsAbi__factory.ts:3]: this is a autogenerated file after using typechain

File: examples/generic-node-example/src/depositGeneric.ts [hunk 18]
```diff
@@ -28,6 +28,8 @@ const depositGeneric = async (): Promise<void> => {
     ColorsABI as ContractInterface
   ) as Colors;
 
+  // const colorContractNode1 = Colors__factory.connect(ColorsAddress.colorsAddressNode1, signerDestinationChain)
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-node-example/src/depositGeneric.ts]: removed

File: examples/generic-colors/public/colors.json
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/public/colors.json:1]: removed

**Comment by wainola** [comment]
> Notice naming for hooks is without prefixed `use` which actually tells developers its a hook Please do this reading https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use

changed

File: examples/generic-colors/src/scripts/deployColors.ts [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+import { promisify } from "util";
+import { ContractInterface, ethers } from "ethers";
+import { NonceManager } from "@ethersproject/experimental";
+import {
+  abi as ColorsABI,
+  bytecode as ColorsByteCode,
+} from "../abis/colors-abi.json";
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+const writeFile = promisify(fs.writeFile);
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/src/scripts/deployColors.ts:11]: done

File: examples/generic-colors/src/hooks/accountData.ts [hunk 20]
```diff
@@ -14,11 +14,21 @@ function AccountData(state: State, dispatch: React.Dispatch<Actions>) {
...
+      payload: colorsAddresses
+    })
+  }
+
 
   useEffect(() => {
     if (state.sygmaInstance !== undefined && state.data !== undefined) {
       console.warn("Getting account info", state.sygmaInstance)
       getData()
+      fetchColors()
```
[the remark this answers is not in the document]
Comment (wainola) [reply examples/generic-colors/src/hooks/accountData.ts]: I have added `json-loader` to config-overrides that allow to import the json file directly

File: examples/generic-node-example/src/depositGeneric.ts (hunk 6 above)
Comment (wainola) [reply examples/generic-node-example/src/depositGeneric.ts]: added type

File: examples/generic-colors/src/App.tsx (hunk 5 above)
Comment (wainola) [reply examples/generic-colors/src/App.tsx]: remove any and adding proper type

**Comment by wainola** [comment]
> > Update: node version of colors example is not going to work until we resolve this [issue](https://github.com/sygmaprotocol/sygma-sdk/issues/196). Frontend examples is properly working with current version of local setup
> 
> if there is an issue with node dependencies, then is better to separate this PR into 2 parts, browser, and node so one part can be merged and another when is blocker resolved. + This will help reviewers to review PR

Since we are going to introduce big changes over the SDK in the weeks to come, we agree on not to fix current examples that are going to be outdated anyways. This example is going to work with current implementation of SDK with Sygma class until we deprecate the Sygma class. Because of this, in the meantime can I we move forward over this PR and merge it @BeroBurny ?

**Comment by BeroBurny** [comment]
> Since we are going to introduce big changes over the SDK in the weeks to come, we agree on not to fix current examples that are going to be outdated anyways. This example is going to work with current implementation of SDK with Sygma class until we deprecate the Sygma class. Because of this, in the meantime can I we move forward over this PR and merge it @BeroBurny ?

If that is the case I don't see the point of merging. What will be achieved if this code is part of Main, considering half of it is not working?

Need opinion on this @MakMuftic @mpetrun5

**Comment by wainola** [comment]
update: this PR is going to be close until we update SDK to new architecture

**Pull request closed without merging**